image/art.ksy

```diff
@@ -0,0 +1,70 @@
+meta:
+  id: arcanum_aart
+  title: "Arcanum: of steamworks and magick obscura graphics"
+  file-extension: art
+  endian: le
+  bit-endian: le
+seq:
+  - id: art_header
+    type: art_header
+  - id: palettes
+    type: art_pal
+    repeat: expr
+    repeat-expr: (art_header.pal_entries[0]>0).to_i+(art_header.pal_entries[1]>0).to_i+(art_header.pal_entries[2]>0).to_i+(art_header.pal_entries[3]>0).to_i

```

Comment:
1. Since `pal_entries` are certain pointers according to the reference [forum thread](https://web.archive.org/web/20160913191941/http://arcanum.game-alive.com/forums/viewtopic.php?f=6&t=53&sid=3758668148aa5a568036d9d589202176), it feels really awkward and unidiomatic to compare them to 0 using `>` (`pal_entries` are fortunately unsigned so it doesn't matter in practice, but if they were signed, it wouldn't even do the right thing). The forum post also includes a code snippet showing how to get the number of palettes and it compares the pointer to `NULL` by equality as I'd expect, and I don't see a reason to change it:

   ```c
   // How to get number of palettes saved in file:
   for (i = 0; i < 4; i++)
   {
      if (header.paletteList[i] != NULL)
         paletteCount++;
   }
   ```

2. Better formatting would be beneficial:

   ```suggestion
       repeat-expr: |
         (art_header.pal_entries[0] != 0).to_i +
         (art_header.pal_entries[1] != 0).to_i +
         (art_header.pal_entries[2] != 0).to_i +
         (art_header.pal_entries[3] != 0).to_i
   ```